src/Environments.jl

```diff
@@ -53,10 +58,56 @@ function StaticEnvironment(
     # create graph
     graph_dict = JSON.parsefile(graph_file)
 
+
     return StaticEnvironment(basin_ids, attributes, graph_dict)
 end
 
+# Some time information of the data
+
+"""
+$(TYPEDEF)
+
+Stores a dated time period with an iterator.
+
+$(TYPEDFIELDS)
+"""
+struct DateWindow
```

Comment:
perhaps this could be defined in a Utils.jl module, since both Environments and States use it?
though I realize that right now our only module is ClimaRivers

Reply:
the function to get upstream basins could also go into utils?

Reply:
Created note in PR message